### PR TITLE
net/raft: split http logic into separate file

### PR DESCRIPTION
**Pull request by jbowens**
Move http related logic into a separate file. raft.go was over
1k lines of code and getting unwieldy.

### CODE REVIEW

**Comment by tessr** [comment]
Is this just copy/paste?

**Comment by jbowens** [comment]
> Is this just copy/paste?

mostly. Service.join used to inline the http request. Just the http-related code from Service.join was copy-pasted into the new requestJoin function. I also made `nodeJoinRequest` and `nodeJoinResponse` named types instead of inlined anonymous types declared twice. Also wrote comments for serveJoin and serveMsg.

File: net/raft/http.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+// nodeJoinResponse is the response format of the /raft/join endpoint
+// used when a new node joins an existing cluster.
+type nodeJoinResponse struct {
+	ID   uint64
+	Snap []byte
+}
+
+// ServeHTTP responds to raft consensus messages at /raft/x,
+// where x is any particular raft internode RPC.
```
Comment (tessr) [inline net/raft/http.go]: supernit: I know this was c/p, but can you remove "particular"?

**Comment by tessr** [comment]
LGTM

**Pull request merged**